### PR TITLE
added support for MACVLAN mode

### PR DESCRIPTION
**Pull request by eyakubovich**

### CODE REVIEW

File: link_linux.go [hunk 1]
```diff
@@ -13,6 +13,14 @@ import (
 var native = nl.NativeEndian()
 var lookupByDump = false
 
+var macvlanModes = [...]uint32{
```
Comment (LK4D4) [inline link_linux.go]: do we really need this? Why not just explicitly cast `MacvlanMode` to `uint32`?

Comment (eyakubovich) [inline link_linux.go]: I'm just following the convention where the constants from Linux headers are in `nl` package. We also need to ensure that 0 value maps to a noop. That's technically possible in this case as the constants start with 1 (for some reason they look like they can be combined in a bitmask), however it seemed a bit magical to do that. But I'm open to getting rid of double constants.

Comment (LK4D4) [inline link_linux.go]: Hehe, actually I already broke that convention in ipvlan patch :) I personally prefer to keep code simpler. But it is up to @vishvananda of course

**Comment by vishvananda** [comment]
One advantage of separating it out is constants are not the same across architectures. Even 386 vs. x86_64 has different constants for syscalls in linux.

**Comment by eyakubovich** [comment]
Other than syscalls, most constants are the same on different architectures. However, it's best to keep them separate for those rare cases.

**Comment by LK4D4** [comment]
I looked at `iproute2` source and noticed, that there is one more mode:
```
MACVLAN_MODE_SOURCE  = 16
```
Also, seems like they use hardcoded values for constants.

**Comment by eyakubovich** [comment]
Added support for MACVLAN_MODE_SOURCE

**Comment by eyakubovich** [comment]
@vishvananda Ping

**Pull request merged**